serializability-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializabilityCheckerCompilerPlugin.scala

```diff
@@ -10,17 +10,27 @@ class SerializabilityCheckerCompilerPlugin(override val global: Global) extends
     """checks whether a specified Akka serialization is applied to all messages, events and persistent state classes"""
 
   //Placeholder options
-  private val pluginOptions = new SerializabilityCheckerOptions(verbose = false)
+  private val pluginOptions = new SerializabilityCheckerOptions(
+    verbose = false,
+    detectionFromGenerics = true,
+    detectionFromGenericMethods = true,
+    detectionFromMethods = true)
 
   override val components: List[PluginComponent] = List(
     new SerializabilityCheckerCompilerPluginComponent(pluginOptions, global))
 
   override def init(options: List[String], error: String => Unit): Boolean = {
-    pluginOptions.verbose = options.contains("verbose")
+    pluginOptions.verbose = options.contains("--verbose")
+    pluginOptions.detectionFromGenerics = !options.contains("--disable-detection-generics")
+    pluginOptions.detectionFromGenericMethods = !options.contains("--disable-detection-generic-methods")
+    pluginOptions.detectionFromMethods = !options.contains("--disable-detection-methods")
     true
   }
 
   override val optionsHelp: Option[String] = Some("""
-      |verbose - print additional info about detected serializability traits and serializable classes
+      |--verbose - print additional info about detected serializability traits and serializable classes
+      |--disable-detection-generics - disables detection of classes using list of generic types
+      |--disable-detection-generic-methods - disables detection of classes using list of methods with generic arguments
+      |--disable-detection-methods - disables detection of classes using list of methods with types
```

Comment:
Does it make sense to split the two? Can you think of a scenario where we'd want to detect from non-generic methods, but not from the generic ones?

Reply:
Each one of them disables a different detection method. I decided to add these flags in case I screwed something up. Instead of disabling the whole plugin, this allows turning one part off.